nhrpd/nhrp_shortcut.c

```diff
@@ -304,6 +305,22 @@ static void nhrp_shortcut_recv_resolution_rep(struct nhrp_reqid *reqid,
 			debugf(NHRP_DEBUG_COMMON,
 			       "Shortcut: no cache for nbma %s", buf[2]);
 		}
+
+		/* Update cache binding for dst_proto as well */
+		if (proto != &pp->dst_proto) {
+			c_dst_proto = nhrp_cache_get(pp->ifp, &pp->dst_proto, 1);
+			if (c_dst_proto) {
+				debugf(NHRP_DEBUG_COMMON,
+			       "Shortcut: cache found, update binding");
+				nhrp_cache_update_binding(c_dst_proto, NHRP_CACHE_DYNAMIC,
+						  holding_time,
+						  nhrp_peer_get(pp->ifp, nbma),
+						  htons(cie->mtu), nbma);
+			} else {
```

Comment:
I could observe that with that patch, an additional dynamic cache entry was made available as shown below. The 3rd entry `10.150.0.1` is an IP located at lan side of remote spoke.
previously, the shortcut created was  ` 10.150.0.0/24` network with a nexthop set to the ip address of the gre address.

Is it expected?
```
show ip nhrp cache 
Iface    Type     Protocol                 NBMA                     Flags  Identity
gre1     local    11.255.255.2             -                               -
gre1     nhs      11.255.255.1             10.125.0.1                T    
gre1     dynamic  11.150.0.1               10.125.0.4                TA   
gre1     dynamic  11.255.255.4             10.125.0.4                TA   
```